### DISCUSSION TITLE
How to use Allure for Karate tests ?

### DISCUSSION THREAD

**Discussion by harmin-parra** [allure-framework/allure2#2580] [Questions & Support] [answered]
Hello

I am aware of the existence of an integration for karate (https://mvnrepository.com/artifact/io.qameta.allure/allure-karate), but I see zero documentation about it :(

Can anybody give me some tips or can point me to a working example ?

Thanks

**Comment by baev** [comment, marked as answer]
There is an official example is available https://github.com/allure-examples/allure-karate-example

**Comment by harmin-parra** [comment]
Thanks a lot :)

I still have this question

Is it possible insert the screenshots in the exact order they are taken during the test instead of putting them at the end of the test?

![image](https://github.com/allure-framework/allure2/assets/16239820/a12a7043-34e0-42e3-aa6c-fd73331eb805)

**Comment by harmin-parra** [comment]
I also realized that some empty tests are added to the report

![image](https://github.com/allure-framework/allure2/assets/16239820/e3d889cd-4b06-4ddb-851a-7e002993722a)

![image](https://github.com/allure-framework/allure2/assets/16239820/c54ccf00-dee6-4bc5-92c7-0636e7afde59)

**Reply by baev** [reply]
Please create an issue in https://github.com/allure-framework/allure-java . Attach a sample repository to reproduce the problem if possible

**Comment by harmin-parra** [comment]
I have another question

How can I add to Allure report the HTTP responses ?

![image](https://github.com/allure-framework/allure2/assets/16239820/d8276547-7cd9-4c3b-b651-3514b6dfa3c9)

![image](https://github.com/allure-framework/allure2/assets/16239820/16096370-9f0d-4724-9c97-33c5d55b37de)

**Reply by baev** [reply]
https://github.com/allure-framework/allure-java/issues/1085
